File: Unreal5/BizXUnreal/Source/BizXUnreal/BizXAAASystems.cpp
#include "BizXAAASystems.h"
void UBizXAAASystems::TickOnce(){ ++Tick; for(auto& N: NPCs){ N.Food=FMath::Max(0.f,N.Food-.005f); N.Rest=FMath::Max(0.f,N.Rest-.003f); N.State=N.Safety<.25f?TEXT("flee"):N.Food<.25f?TEXT("seek_food"):N.Rest<.2f?TEXT("sleep"):TEXT("work"); } }
float UBizXAAASystems::Market(const FString& Good,float Supply,float Demand){ const float P=FMath::Max(.01f,(Demand+1.f)/(Supply+1.f)); Prices.Add(Good,P); return P; }
int32 UBizXAAASystems::ResolveAbility(int32 BaseDamage,int32 Defense,int32 Roll) const { return Roll+5>=Defense ? BaseDamage*(Roll==20?2:1):0; }
void UBizXAAASystems::TriggerWorldEvent(const FString& EventId){ ActiveEvents.AddUnique(EventId); }

File: Unity3D/Assets/BizX/Game/BizXAAASystems.cs
using System; using System.Collections.Generic; using UnityEngine;

namespace BizX.Game {
[Serializable] public class BizXWorldCell { public string id; public long seed; public string biome="plains"; public int population; public string weather="clear"; public bool loaded; }
[Serializable] public class BizXNPC { public string id; public string faction; public float food=1, rest=1, safety=1; public string state="idle"; }
[Serializable] public class BizXVehicle { public string id; public string owner; public float fuel=100, durability=100, speed=1; public int cargo, cargoCapacity=10; }
public sealed class BizXAAASystems : MonoBehaviour {
 public int tick; public List<BizXWorldCell> cells=new(); public List<BizXNPC> npcs=new(); public List<BizXVehicle> vehicles=new(); public List<string> activeEvents=new(); public Dictionary<string,float> prices=new();
 public void TickOnce(){tick++; foreach(var n in npcs){n.food=Mathf.Max(0,n.food-.005f);n.rest=Mathf.Max(0,n.rest-.003f);n.state=n.safety<.25f?"flee":n.food<.25f?"seek_food":n.rest<.2f?"sleep":"work";}}
 public float Market(string good,float supply,float demand){float p=Mathf.Max(.01f,(demand+1)/(supply+1));prices[good]=p;return p;}
 public int ResolveAbility(int baseDamage,int defense,int roll){bool hit=roll+5>=defense;return hit?baseDamage*(roll==20?2:1):0;}
 public void TriggerWorldEvent(string id){if(!activeEvents.Contains(id))activeEvents.Add(id);}
}
}
